offchain-modules/packages/x/src/handlers/eth.ts

```diff
@@ -77,9 +77,14 @@ export class EthHandler {
   }
 
   async watchNewBlock() {
-    await this.ethChain.watchNewBlock(this.lastHandledBlockHeight, async (newBlock: ethers.providers.Block) => {
-      await this.onBlock(newBlock);
-    });
+    try {
```

Comment:
replace watchNewBlock in contract.ts

```suggestion
    
   async watchNewBlock(startHeight: number, handleBlockFunc) {
    let block: ethers.providers.Block;
    for (let height = startHeight + 1; ; ) {
      while (true){
        try {
          block = await this.provider.getBlock(height);
          if (!block) {
            //wait for new block
            await asyncSleep(5 * 1000);
          }else{
            break;
          }
        }catch (e) {
          logger.error(`Eth watchNewBlock blockHeight:${height} error:${e.message}`)
          await asyncSleep(3 * 1000);
        }
      }
      await handleBlockFunc(block);
      height++;
    }
  }
```